src/pages/wallet/WalletStatementPage.tsx

```diff
@@ -1,61 +1,41 @@
 import {format, getMonth, getYear} from 'date-fns';
 import Str from 'expensify-common/lib/str';
-import lodashGet from 'lodash/get';
-import PropTypes from 'prop-types';
 import React, {useEffect} from 'react';
 import {withOnyx} from 'react-native-onyx';
+import type {OnyxEntry} from 'react-native-onyx';
 import FullPageOfflineBlockingView from '@components/BlockingViews/FullPageOfflineBlockingView';
 import HeaderWithBackButton from '@components/HeaderWithBackButton';
-import networkPropTypes from '@components/networkPropTypes';
-import {withNetwork} from '@components/OnyxProvider';
 import ScreenWrapper from '@components/ScreenWrapper';
 import WalletStatementModal from '@components/WalletStatementModal';
-import withLocalize, {withLocalizePropTypes} from '@components/withLocalize';
+import useLocalize from '@hooks/useLocalize';
+import useNetwork from '@hooks/useNetwork';
 import useThemePreference from '@hooks/useThemePreference';
-import compose from '@libs/compose';
 import DateUtils from '@libs/DateUtils';
 import fileDownload from '@libs/fileDownload';
 import Growl from '@libs/Growl';
 import Navigation from '@libs/Navigation/Navigation';
+import type {WalletStatementNavigatorParamList} from '@navigation/types';
 import * as User from '@userActions/User';
 import CONFIG from '@src/CONFIG';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import type {WalletStatement} from '@src/types/onyx';
 
-const propTypes = {
-    /** The route object passed to this page from the navigator */
-    route: PropTypes.shape({
-        /** Each parameter passed via the URL */
-        params: PropTypes.shape({
-            /** The statement year and month as one string, i.e. 202110 */
-            yearMonth: PropTypes.string.isRequired,
-        }).isRequired,
-    }).isRequired,
-
-    walletStatement: PropTypes.shape({
-        /** Whether we are currently generating a PDF version of the statement */
-        isGenerating: PropTypes.bool,
-    }),
-
-    /** Information about the network */
-    network: networkPropTypes.isRequired,
-
-    /** Indicates which locale the user currently has selected */
-    preferredLocale: PropTypes.string,
-
-    ...withLocalizePropTypes,
+type WalletStatementOnyxProps = {
+    walletStatement: OnyxEntry<WalletStatement>;
 };
 
-const defaultProps = {
-    walletStatement: {
-        isGenerating: false,
-    },
-    preferredLocale: CONST.LOCALES.DEFAULT,
-};
+type WalletStatementPageProps = WalletStatementOnyxProps & StackScreenProps<WalletStatementNavigatorParamList, typeof SCREENS.WALLET_STATEMENT_ROOT>;
 
-function WalletStatementPage(props) {
+function WalletStatementPage({walletStatement, route}: WalletStatementPageProps) {
     const themePreference = useThemePreference();
-    const yearMonth = lodashGet(props.route.params, 'yearMonth', null);
+    const yearMonth = route.params?.yearMonth ?? null;
```

Comment:
```suggestion
    const yearMonth = route.params.yearMonth ?? null;
```

Could you check if `route.params` can be really `undefined`?

Reply:
removed